packages/stream-model-instance/src/model-instance-document.ts

```diff
@@ -222,6 +242,32 @@ export class ModelInstanceDocument<T = Record<string, any>> extends Stream {
       prev: this.tip,
       id: this.id.cid,
     }
+    // Null check is necessary to avoid `undefined` value that can't be encoded with IPLD
+    if (header != null) {
+      rawCommit.header = header
+    }
+    const commit = await ModelInstanceDocument._signDagJWS(this.api, rawCommit)
+    const updated = await this.api.applyCommit(this.id, commit, options)
+    this.state$.next(updated.state)
+  }
+
+  /**
+   * Set the index metadata field for the stream
+   * @param shouldIndex - Whether the stream should be indexed or not
+   * @param opts - Additional options
+   */
+  async shouldIndex(shouldIndex: boolean, opts: CommonUpdateOpts = {}): Promise<void> {
+    // No-op if the wanted value is the current one
+    if (this.metadata.shouldIndex === shouldIndex) {
+      return
+    }
+
+    const rawCommit: RawCommit = {
```

Comment:
nitpick, can implement `shouldIndex()` in terms of `replace()` or `patch()` and avoid some code duplication